user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQIInputArguments.md

```diff
@@ -1,15 +1,33 @@
 ---
-uid: IGQIInputArguments
+uid: GQI_IGQIInputArguments
 ---
 
-# IGQIInputArguments
+# IGQIInputArguments Interface
 
-The *IGQIInputArguments* interface can be used to have the user specify an argument, for example the CSV file from which data should be parsed, or a filter that should be applied. This interface has the following methods:
+## Definition
 
-| Method | Input arguments | Output arguments | Description |
-|--|--|--|--|
-| GetInputArguments | - | GQIArgument[] | Asks for additional information from the user when the data source is configured. |
-| OnArgumentsProcessed | OnArgumentsProcessedInputArgs | OnArgumentsProcessedOutputArgs | Event to indicate that the arguments have been processed. |
+Namespace: `Skyline.DataMiner.Analytics.GenericInterface`  
+Assembly: `SLAnalyticsTypes.dll`
 
-> [!NOTE]
+Implementing this interface makes it possible to retrieve input from the user through arguments.
+
+> [!IMPORTANT]
 > The GQI does not validate the input arguments specified by the user. For example, a user can input an SQL query as a string input argument, and the content of the string argument will be forwarded to the ad hoc data source implementation without validation.
+
+## Methods
+
+### GQIArgument[] GetInputArguments()
+
+The GQI will request the arguments.
+
+#### Returns
+
+An array of [GQIArgument](xref:GQI_GQIArgument), which provides all arguments the user can fill in.
+
+### void OnArgumentsProcessed(OnArgumentsProcessedInputArgs args)
+
+The user has filled in the arguments.
```

Comment:
This is a bit misleading, since it's also just called when the query is executed without a user filling in arguments again.